File: src/dataset/c/triangle_exercism-viettronics868/claude/ClaudeSonnet4_triangle_v3.c
#include "triangle.h"
#include <stdio.h>
#include <stdbool.h>

bool is_triangle(triangle_t input){
    return (input.a > 0 && input.b > 0 && input.c > 0) &&
           (input.a + input.b > input.c) &&
           (input.a + input.c > input.b) &&
           (input.b + input.c > input.a);
}

bool is_scalene(triangle_t input){
    return is_triangle(input) &&
           (input.a != input.b) &&
           (input.a != input.c) &&
           (input.b != input.c);
}

bool is_isosceles(triangle_t input){
    if (!is_triangle(input)) return false;
    
    int equal_sides = 0;
    if (input.a == input.b) equal_sides++;
    if (input.a == input.c) equal_sides++;
    if (input.b == input.c) equal_sides++;
    
    return equal_sides >= 1;
}

bool is_equilateral(triangle_t input){
    return is_triangle(input) &&
           (input.a == input.b) &&
           (input.b == input.c);
}

File: src/dataset/Java/triangle_exercism-uzilan/claude/ClaudeSonnet4_Triangle_v3.java
import static java.lang.Math.max;

class Triangle {

    private final byte distinctSides;

    Triangle(double side1, double side2, double side3) throws TriangleException {
        final double max = max(side1, max(side2, side3));
        if (max * 2 >= side1 + side2 + side3) {
            throw new TriangleException();
        }

        byte count = 1;
        if (side1 != side2) count++;
        if (side1 != side3 && side2 != side3) count++;
        
        distinctSides = count;
    }

    boolean isEquilateral() {
        return distinctSides == 1;
    }

    boolean isIsosceles() {
        return distinctSides <= 2;
    }

    boolean isScalene() {
        return distinctSides == 3;
    }
}